### DISCUSSION TITLE
Text's not showing up

### DISCUSSION THREAD

**Discussion by dman-os** [emilk/egui#327] [Q&A]
I'm sure this is a bug in my integration but I have no where else to turn and I've been banging my head against this for days. I'm asking here hoping that'd someone would me point me in the right direction.

Anyways, I tried integrating egui into my app which uses `winit` and `wgpu` but something went wrong and I'm unable to see text.

The following snippet...
```rust
egui::Window::new("stuff")
        .resizable(true)
        .show(&egui_cmds, |ui| {
            ui.label("Bonjurno");
            ui.label("Bonjurno");
            ui.label("Bonjurno");
            ui.label("Bonjurno");
            ui.label("Bonjurno");
            ui.label("Bonjurno");
            ui.label("Bonjurno");
        });
    egui::Window::new("another stuff")
        .resizable(true)
        .show(&egui_cmds, |ui| {
            ui.label("hellow egui");
            ui.label("hellow egui");
            ui.label("hellow egui");
            ui.label("hellow egui");
            ui.label("hellow egui");
            ui.label("hellow egui");
            ui.label("hellow egui");
        });
    egui::CentralPanel::default().show(&egui_cmds, |ui| {
        ui.label("Hello world!");
        if ui.button("Click me").clicked() {
            deps::tracing::info!("clicked");
        }
    });
```
... gives me the following output.
![image](https://user-images.githubusercontent.com/61868957/115280978-2f204900-a151-11eb-99d8-8d7204cde84f.png)

I'm able to drag the windows around and double clicking on them minimizes them but the resize handles don't work.

My integration was based on the ones by [hasenbanck](https://github.com/hasenbanck/egui_wgpu_backend) linked in the `readme` which I had to modify for use in an ECS.

I've tried disabling backface culling, checked if the font texture is loaded properly in renderdoc (it is), checked if the meshes are being loaded properly in renderdoc (they appear to be) and tried a bunch of other things. I come here in desperation and in hopes that someone might give me ideas for things to try.

**Comment by dman-os** [comment]
This problem resolved itself after recent updates in `wgpu` or one of it's dependencies. Coo coo cool.

**Reply by VoilaNeighbor** [reply]
:(

I was using `egui-glow` and I cannot get text working, either.

**Comment by nemtbar** [comment]
I'm using v0.32.2 of both egui-winit, and egui-wgpu, and I can't get the text to show up either, the egui docs say that I should call `Self::begin_pass()` and `Self::font_image_delta()` every frame on the fonts of the context, but I tried that, and it didn't work, and I believe that these are handled by egui-wgpu.


    fn  egui_render(&mut self) {
        let Self::Ready {gpu, window, surface, myUi, .. } = self else {return;};
        let input = myUi.state.take_egui_input(window);

        let full_output = myUi.ctx.run(input, |ctx| {
            ctx.set_visuals(egui::Visuals::dark());
            egui::CentralPanel::default().show(&ctx, |ui| {
                if ui.button("click me!!!!").clicked() {
                    println!("whoaaaaa");
                }
            });
        });

        
        let clipped = myUi.ctx.tessellate(full_output.shapes, full_output.pixels_per_point);

        for (id, delta) in &full_output.textures_delta.set {
            myUi.renderer.update_texture(&gpu.device, &gpu.queue, *id, delta);
        }

        let mut encoder = gpu.device.create_command_encoder(&wgpu::CommandEncoderDescriptor {
            label: Some("ui encoder")
        });

        let screen_desc = egui_wgpu::ScreenDescriptor {
            size_in_pixels: window.inner_size().into(),
            pixels_per_point: window.scale_factor() as f32
        };

        myUi.renderer.update_buffers(&gpu.device, &gpu.queue, &mut encoder, &clipped, &screen_desc);


        let view = match surface.get_current_texture() {
            Ok(t) => t,
            Err(wgpu::SurfaceError::Lost) => {
                surface.configure(&gpu.device, &gpu.config);
                return;
            }
            Err(e) => {
                eprint!("surface error:\n{}", e);
                return;
            }
        };
    
        let rpass = encoder.begin_render_pass(&wgpu::RenderPassDescriptor {
            label: Some("ui renderpass"),
            color_attachments: &[Some(wgpu::RenderPassColorAttachment {
                view: &view.texture.create_view(&wgpu::TextureViewDescriptor::default()),
                resolve_target: None,
                ops: wgpu::Operations { 
                    load: wgpu::LoadOp::Clear(wgpu::Color::RED),
                    store: wgpu::StoreOp::Store
                }
            })],
            ..Default::default()
        });

        myUi.renderer.render(&mut rpass.forget_lifetime(), &clipped,  &screen_desc );
        

        gpu.queue.submit(Some(encoder.finish()));

        view.present();
        for id in &full_output.textures_delta.free {
            myUi.renderer.free_texture(id);
        }
    

    }

**Reply by nemtbar** [reply]
I found out that you have to use egui and wgpu itself, not just the re-exports of the egui-wgpu and egui-winit crate. This helped me a lot: [https://github.com/hasenbanck/egui_example](url)
